### ISSUE TITLE
node.get("style", "") may return None

### ISSUE THREAD

**Issue by rumpeltux** [inkstitch/inkstitch#979]
I got this error on the current master branch with inkscape 1.0.1:

    Ink/Stitch experienced an unexpected error.
    If you'd like to help, please file an issue at https://github.com/inkstitch/inkstitch/issues and include the entire error description below: 

    Traceback (most recent call last):
      File "inkstitch.py", line 44, in <module>
        extension.affect(args=remaining_args)
      File "/usr/share/inkscape/extensions/inkex/deprecated.py", line 181, in affect
        return self.run(args=args)
      File "/usr/share/inkscape/extensions/inkex/base.py", line 140, in run
        self.save_raw(self.effect())
      File "lib/extensions/params.py", line 590, in effect
        frame = SettingsFrame(tabs_factory=self.create_tabs, on_cancel=self.cancel)
      File "lib/extensions/params.py", line 324, in __init__
        self.tabs = self.tabs_factory(self.notebook)
      File "lib/extensions/params.py", line 555, in create_tabs
        nodes_by_class = self.get_nodes_by_class()
      File "lib/extensions/params.py", line 485, in get_nodes_by_class
        nodes = self.get_nodes()
      File "lib/extensions/base.py", line 176, in get_nodes
        return self.descendants(self.document.getroot(), troubleshoot=troubleshoot)
      File "lib/extensions/base.py", line 165, in descendants
        nodes.extend(self.descendants(child, selected, troubleshoot))
      File "lib/extensions/base.py", line 165, in descendants
        nodes.extend(self.descendants(child, selected, troubleshoot))
      File "lib/extensions/base.py", line 147, in descendants
        if element.has_style('display') and element.get_style('display') is None:
      File "lib/elements/element.py", line 184, in has_style
        return self._get_style_raw(style_name) is not None
      File "/usr/lib/python2.7/dist-packages/backports/functools_lru_cache.py", line 113, in wrapper
        result = user_function(*args, **kwds)
      File "lib/elements/element.py", line 167, in _get_style_raw
        style = self.parse_style()
      File "/usr/lib/python2.7/dist-packages/backports/functools_lru_cache.py", line 113, in wrapper
        result = user_function(*args, **kwds)
      File "lib/elements/element.py", line 158, in parse_style
        declarations = tinycss2.parse_declaration_list(node.get("style", ""))
      File ".local/lib/python2.7/site-packages/tinycss2/parser.py", line 176, in parse_declaration_list
        tokens = _to_token_iterator(input, skip_comments)
      File ".local/lib/python2.7/site-packages/tinycss2/parser.py", line 21, in _to_token_iterator
        return iter(input)
    TypeError: 'NoneType' object is not iterable

This seems to happen when there’s a \<g\> node, just grouping without any styles.

I was able to fix it with this patch:
```diff
diff --git a/lib/elements/element.py b/lib/elements/element.py
index fe1491d8..12f5e9f9 100644
--- a/lib/elements/element.py
+++ b/lib/elements/element.py
@@ -155,7 +155,14 @@ class EmbroideryElement(object):
     def parse_style(self, node=None):
         if node is None:
             node = self.node
-        declarations = tinycss2.parse_declaration_list(node.get("style", ""))
+        style = node.get("style", "")
+        if style is None: style=""
+        declarations = tinycss2.parse_declaration_list(style)
         style = {declaration.lower_name: declaration.value[0].serialize() for declaration in declarations}
         return style

```

**Comment by kaalleen** [comment]
Thank you (again :) ). We'll have a look into it.

**Comment by kaalleen** [comment]
I believe this is already fixed within the update branch (#880).
Thank you for offering a solution, we really appreciate your help.

**Comment by rumpeltux** [comment]
I can confirm that this is fixed in the inkscape1.0 branch.
